### PR TITLE
OTP-UI Icons

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR removes icons from lib/components/icons and uses those from the OTP-UI icons package instead. The PR also removes unused narrative profile components.

EDITS: Please see the Remarks section below for edits and testing notes.

### Remarks

1. [Outdated as of 9219719bac7f92f1d7ead440e50f9e218947b3d5] Full support of custom icons in OTP-ui is pending opentripplanner/otp-ui#119, however the refactor is mostly compatible with the currently released OTP-ui library.
2. [Outdated as of 9219719bac7f92f1d7ead440e50f9e218947b3d5] In  https://github.com/opentripplanner/otp-react-redux/blob/02fef2765dd684acadfbdcf0d1676853a0cdd4db/example.js#L30, there is a `MyModeIcon` variable passed to various `ModeIcon` props, however `ModeIcon` props will take effect pending a change in `SettingsSelectorPanel` in opentripplanner/otp-ui#119.
3. [Outdated as of 6c10ea651c951d0497289ac8111d7adf867dedc1] As a result, the default icon for "Take Transit" in OTP-RR is changed, and there are subtle visual changes to the itinerary narratives as shown in the screenshots below.
(Note that the bus and walk icon sizes are pending OTP release per https://github.com/opentripplanner/otp-ui/pull/105 to address [this comment](https://github.com/opentripplanner/otp-react-redux/pull/142#pullrequestreview-383985093)) Beware, visual changes taking place! (see screenshots below)
 
image

image

screenshot if using LineItinerary
image

4. As of 6c10ea651c951d0497289ac8111d7adf867dedc1, visual styles for trip form elements (not the map or nav bar) match newplanner.trimet.org, and the old styling is abandoned.

### CODE REVIEW

File: lib/components/map/default-map.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
   setMapPopupLocation,
   setMapPopupLocationAndGeocode
 } from '../../actions/map'
+
```
Comment (evansiroky) [inline lib/components/map/default-map.js:16]: No space needed here.

File: lib/components/narrative/default/itinerary-summary.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 
-import ModeIcon from '../../icons/mode-icon'
+import TriMetLegIcon from '@opentripplanner/icons/lib/trimet-leg-icon'
```
Comment (evansiroky) [inline lib/components/narrative/default/itinerary-summary.js:4]: Should be at the very least grouped with the 3rd party imports and ideally sorted before React and PropTypes. Also PropTypes should be before React.

**Comment by evansiroky** [comment]
Just noticed that this PR depends on #147. I'm going to unassign myself until that one gets merged into dev.

**Comment by binh-dam-ibigroup** [comment]
There are some components I deleted an need to bring back... I will reassign you when done.

**Comment by evansiroky** [comment]
Unassigning myself again until #147 is merged.

File: lib/components/form/default-search-form.js [hunk 3]
```diff
@@ -1,10 +1,16 @@
-import React, { Component } from 'react'
+import TriMetModeIcon from '@opentripplanner/icons/lib/trimet-mode-icon'
```
Comment (landonreed) [inline lib/components/form/default-search-form.js:1]: I think we should add a new generic `ModeIcon` to otp-ui that doesn't use the TriMet-specific icons, but instead uses the ones that were originally used in this repo. One reason: we don't want all the rail icons to appear as the Max icon. Another: it is likely confusing to sprinkle in mentions of `TriMet`-branded things in this library, especially with icons which can be closely tied to agency-specific branding.

File: lib/components/form/default-search-form.js [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
-import SwitchButton from './switch-button'
 import TabbedFormPanel from './tabbed-form-panel'
-import defaultIcons from '../icons'
+import SwitchButton from './switch-button'
+
+// Use the icons props to override icons for certain modes in the mode selector panel.
+// If no icon is provided for a specific mode, the default OTP-ui icon will be used.
+const customIcons = {
+  TRANSIT: <TriMetModeIcon label='rail' />
+}
```
Comment (landonreed) [inline lib/components/form/default-search-form.js:13]: Default used to be `bus`, right? Why the change?

File: lib/components/form/default-search-form.js [hunk 5]
```diff
@@ -13,7 +19,7 @@ export default class DefaultSearchForm extends Component {
   }
 
   static defaultProps = {
-    icons: defaultIcons,
+    icons: customIcons,
```
Comment (landonreed) [inline lib/components/form/default-search-form.js:22]: Just from a variable naming perspective, it feels wrong to set a default to a variable called `customIcons`.

File: lib/components/narrative/narrative-routing-results.js [hunk 6]
```diff
@@ -32,12 +31,8 @@ class NarrativeRoutingResults extends Component {
...
 
     return (
-      routingType === 'ITINERARY'
-        ? <TabbedItineraries itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} itineraries={itineraries} customIcons={customIcons} />
-        : <NarrativeProfileOptions
-          itineraryClass={itineraryClass}
-          customIcons={customIcons}
-        />
+      routingType === 'ITINERARY' &&
+        <TabbedItineraries itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} itineraries={itineraries} customIcons={customIcons} />
```
Comment (landonreed) [inline lib/components/narrative/narrative-routing-results.js:35]: Probably warrants putting this on multiple lines (seems to violate the characters per line limit).

File: lib/components/form/default-search-form.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/default-search-form.js:1]: I agree. We started off with a TriMet-focused approach... How about adding the generic icons and a generic mode-icon/leg-icon class to OTP-ui and isolate the TriMet-specific icons inside OTP-ui?

**Comment by landonreed** [comment]
Also, I noticed a bug: when there are two car rental mode options and one is selected, both are highlighted:
image

`config.yml` (modes only):
```
  accessModes:
    - mode: BICYCLE
      label: Transit + Personal bike
    - mode: BICYCLE_RENT
      label: Transit + Biketown
      company: BIKETOWN
    - mode: MICROMOBILITY_RENT
      label: Transit + E-scooter rental
    - mode: CAR_PARK
      label: Park & Ride
    - mode: CAR_RENT
      label: Transit + car2go
      company: CAR2GO
    - mode: CAR_RENT
      label: Transit + REACH NOW
      company: REACHNOW
    - mode: CAR_HAIL
      label: Transit + Uber
      company: UBER
      showWheelchairSetting: true
  exclusiveModes:
    - WALK
    - BICYCLE
```

File: lib/components/narrative/line-itin/itin-summary.js [hunk 7]
```diff
@@ -82,6 +82,19 @@ export default class ItinerarySummary extends Component {
     itinerary: PropTypes.object
   }
 
+  getIcon = (iconId, customIcons) => {
+    // Check if there is a custom icon
+    if (customIcons && iconId in customIcons) {
```
Comment (evansiroky) [inline lib/components/narrative/line-itin/itin-summary.js:87]: Is there a way we can get rid of this `customIcons` variable altogether? It would be nice to have everything inside the mode icon component.

File: lib/components/form/default-search-form.js (hunk 3 above)
Comment (evansiroky) [reply lib/components/form/default-search-form.js:1]: Seems like a good idea to have a generic library specifically for otp-react-redux since we use it in more regions than just Portland, OR.

File: lib/components/narrative/line-itin/itin-summary.js (hunk 7 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/line-itin/itin-summary.js:87]: Working on it. I will shortly push a PR against OTP-ui to address that.

**Comment by binh-dam-ibigroup** [comment]
@landonreed The issues with mode selection you are mentioning above are handled by https://github.com/opentripplanner/otp-ui/pull/119.

File: lib/components/narrative/line-itin/itin-summary.js (hunk 7 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/line-itin/itin-summary.js]: This is resolved with opentripplanner/otp-ui#119.

File: lib/components/form/default-search-form.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/default-search-form.js]: The above topic is handled with opentripplanner/otp-ui#119 and the latest commits to this PR.

File: example.js [hunk 8]
```diff
@@ -90,14 +94,15 @@ class OtpRRExample extends Component {
 
     /** mobile view **/
     const mobileView = (
-      <MobileMain map={(<Map />)} title={(<div className='navbar-title'>OpenTripPlanner</div>)} />
+      <MobileMain LegIcon={MyLegIcon} ModeIcon={MyModeIcon} map={(<Map />)} title={(<div className='navbar-title'>OpenTripPlanner</div>)} />
```
Comment (landonreed) [inline example.js:97]: Let's put this on multiple lines.

File: lib/components/mobile/main.js [hunk 9]
```diff
@@ -99,10 +99,10 @@ class MobileMain extends Component {
         return <MobileDateTimeScreen />
 
       case MobileScreens.SET_OPTIONS:
-        return <MobileOptionsScreen icons={icons} />
+        return <MobileOptionsScreen ModeIcon={ModeIcon} />
 
       case MobileScreens.RESULTS_SUMMARY:
-        return <MobileResultsScreen map={map} itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} icons={icons} />
+        return <MobileResultsScreen map={map} itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} LegIcon={LegIcon} />
```
Comment (landonreed) [inline lib/components/mobile/main.js:105]: multiple lines

File: lib/components/mobile/results-screen.js [hunk 10]
```diff
@@ -95,7 +95,7 @@ class MobileResultsScreen extends Component {
   _toggleRealtime = () => this.props.setUseRealtimeResponse({useRealtime: !this.props.useRealtime})
 
   render () {
-    const { error, icons, itineraryClass, itineraryFooter, query, realtimeEffects, resultCount, useRealtime, activeItineraryIndex } = this.props
+    const { error, itineraryClass, itineraryFooter, LegIcon, query, realtimeEffects, resultCount, useRealtime, activeItineraryIndex } = this.props
```
Comment (landonreed) [inline lib/components/mobile/results-screen.js:98]: multiple lines

File: example.js [hunk 11]
```diff
@@ -26,6 +27,9 @@ import {
 // load the OTP configuration
 import otpConfig from './config.yml'
 
+const MyLegIcon = TriMetLegIcon // For testing, you can use () => <Ferry />
+const MyModeIcon = () => TriMetModeIcon // For testing, you can use () => <AerialTram />
```
Comment (landonreed) [inline example.js]: Can you add commented out lines below this that people can easily uncomment to try out the testing overrides? Also, it would be good to describe what the testing will show (and where).

File: example.js [hunk 12]
```diff
@@ -27,8 +27,21 @@ import {
...
+/**
+ * For testing, try uncommenting the following two statements (and comment the two above),
+ * and see how the icons get changed in:
+ * - the mode options panel (select transit, bike+transit, etc.)
+ * - the itinerary narrative (step-by-step directions).
+ *
+ * For more advanced testing, you can replicate and customize components and observe the change in icons.
+ * - For LegIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-leg-icon.js
+ * - For ModeIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-mode-icon.js
+ */
```
Comment (landonreed) [inline example.js:42]: Fantastico!

**Comment by binh-dam-ibigroup** [comment]
@landonreed per a suggestion by @evansiroky, I removed the `customIcons` support but I realized we might still need some folks to occasionally override the std transit or streetcar icons, for instance without having to define a whole new `ModeIcon` or `LegIcon`. Let me think through it.

**Comment by binh-dam-ibigroup** [comment]
@landonreed and @evansiroky, I have updated the opentripplanner/otp-ui packages and updated the examples for custom icon sets accordingly, so you can review.

File: example.js [hunk 13]
```diff
@@ -28,21 +38,56 @@ import {
...
+    leg.routeLongName.startsWith('MAX')
+  ) {
+    return <CustomStreetcarIcon />
+  } else if (leg.rentedBike) {
+    return <CustomBikeRentalIcon />
+  }
+  return <LegIcon leg={leg} ModeIcon={MyModeIcon} {...props} />
+}
+*/
+
```
Comment (landonreed) [inline example.js:90]: @binh-dam-ibigroup, this is growing to be a lot to include in the `example.js` file. I think it might make more sense to move this to a `custom-icons.js` file and enable playing around with icon overrides with a single change. I've provided some changes in pursuit of this in https://github.com/opentripplanner/otp-react-redux/pull/164.

Comment (binh-dam-ibigroup) [reply example.js:81]: Good idea.

File: lib/components/mobile/results-screen.js [hunk 14]
```diff
@@ -95,7 +95,17 @@ class MobileResultsScreen extends Component {
...
+    const {
+      error,
+      itineraryClass,
+      itineraryFooter,
+      LegIcon,
+      query,
+      realtimeEffects,
+      resultCount,
+      useRealtime,
+      activeItineraryIndex
```
Comment (evansiroky) [inline lib/components/mobile/results-screen.js:107]: So close to perfect alphabetization until here.
